pythonmod/pythonmod.c

```diff
@@ -547,15 +561,15 @@ void pythonmod_deinit(struct module_env* env, int id)
       Py_XDECREF(pe->data);
       PyGILState_Release(gil);
 
-      if(--py_mod_count==0) {
-         PyEval_RestoreThread(mainthr);
-         Py_Finalize();
-         mainthr = NULL;
-      }
+      py_mod_count--;
    }
    pe->fname = NULL;
    free(pe);
 
+   /* iterate over all possible callback types and clean up each in turn */
+   for (int cbtype = 0; cbtype < inplace_cb_types_total; cbtype++)
+      inplace_cb_delete(env, cbtype, id);
+
```

Comment:
Good catch! It seems the callback delete functions are not exposed to Python.
Since the list of callbacks is really small (or non-existent) this works fine and keeps it simple for Python module users.
If we have more in the future we can revisit this part (well, together with the callback structures that would need to be something more than just a list).
But I would prefer `cbtype` to be declared at the start of the function like most Unbound code.
```suggestion
   /* iterate over all possible callback types and clean up each in turn */
   for(cbtype = 0; cbtype < inplace_cb_types_total; cbtype++)
      inplace_cb_delete(env, cbtype, id);

```

Reply:
> Good catch! It seems the callback delete functions are not exposed to Python. Since the list of callbacks is really small (or non-existent) this works fine and keeps it simple for Python module users. If we have more in the future we can revisit this part (well, together with the callback structures that would need to be something more than just a list).

Agreed.

> But I would prefer `cbtype` to be declared at the start of the function like most Unbound code.

Done. Thanks